### PR TITLE
Removed PHP 5 work arounds and fix PHP 8 call_user_* issues

### PR DESCRIPTION
**Pull request by GrahamCampbell**
// cc @gmponos this safely removes `call_user_func_array` too

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
The phpstan warning seems to be a bug in phpstan. I'll re-gen the baseline and report the bug to them.

**Comment by GrahamCampbell** [comment]
https://github.com/phpstan/phpstan/issues/3559

File: src/Handler/StreamHandler.php [hunk 1]
```diff
@@ -551,25 +549,24 @@ static function () use ($ident, $value, $map, $args): void {
         );
     }
 
-    private function addNotification(array &$params, callable $notify): void
+    private static function addNotification(array &$params, callable $notify): void
```
Comment (gmponos) [inline src/Handler/StreamHandler.php:552]: what's the benefit of having them static? is this for some perf. reasons?

File: src/Handler/MockHandler.php [hunk 2]
```diff
@@ -75,7 +75,8 @@ public function __construct(
         $this->onRejected = $onRejected;
 
         if ($queue) {
-            \call_user_func_array([$this, 'append'], $queue);
+            // array_values included for BC
+            $this->append(...array_values($queue));
```
Comment (gmponos) [inline src/Handler/MockHandler.php:79]: Nice workaround.

File: src/Handler/StreamHandler.php (hunk 1 above)
Comment (GrahamCampbell) [reply src/Handler/StreamHandler.php:552]: Yeh, I just noticed these private functions could be made static. This also makes them clearer to the programmer that they won't have any side effects on the instance.

**Comment by GrahamCampbell** [comment]
NB This PR also fixes issues with PHP 8, since `call_user_func_array` now gives array keys meaning, and we don't want that.

**Comment by GrahamCampbell** [comment]
For that reason, and chance we could sneak this into 7.1.0 @Nyholm?

**Review by Nyholm — APPROVED** [review]
This looks good. 

No, lets not wedge more things into 7.1. I've looked that version in =)
I would be more happy to release a 7.2 or 7.1.x with this later this month

**Comment by GrahamCampbell** [comment]
Cool, just rebased this to fix conflicts. Shame about the false positive from phpstan. https://3v4l.org/p3XfW

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**